src/features/click-to-load.js

```diff
@@ -1699,48 +1707,34 @@ function createYouTubePreview (originalElement, widget) {
     return { youTubePreview, shadowRoot }
 }
 
-// Convention is that each function should be named the same as the sendMessage
-// method we are calling into eg. calling `sendMessage('getClickToLoadState')`
-// will result in a response routed to `updateHandlers.getClickToLoadState()`.
-const messageResponseHandlers = {
-    getClickToLoadState (response) {
-        devMode = response.devMode
-        isYoutubePreviewsEnabled = response.youtubePreviewsEnabled
-
-        // Mark the feature as ready, to allow placeholder replacements to
-        // start.
-        readyToDisplayPlaceholdersResolver()
-    },
-    setYoutubePreviewsEnabled (response) {
-        if (response?.messageType && typeof response?.value === 'boolean') {
-            originalWindowDispatchEvent(
-                createCustomEvent(
-                    response.messageType, { detail: response.value }
-                )
-            )
-        }
-    },
-    getYouTubeVideoDetails (response) {
-        if (response?.status && typeof response.videoURL === 'string') {
-            originalWindowDispatchEvent(
-                createCustomEvent(
-                    'ddg-ctp-youTubeVideoDetails',
-                    { detail: response }
-                )
-            )
+export default class ClickToLoad extends ContentFeature {
+    // Messaging layer between Click to Load and the Platform
+    get messaging () {
+        if (this._messaging) return this._messaging
+
+        if (this.platform.name === 'android' || this.platform.name === 'extension') {
+            const messagingContext = new MessagingContext({
+                context: 'contentScopeScripts',
+                featureName: 'click-to-load',
+                env: this.isDebug ? 'development' : 'production'
+            })
+            const messagingConfig = new SendMessageMessagingConfig()
+            this._messaging = new Messaging(messagingContext, messagingConfig)
+            _messagingModuleScope = this._messaging
```

Comment:
I should actually move this to inside `init()`, right after we check for`this.messaging`, to ensure it is initialised right away.

Reply:
I think this should be sync so shouldn't need to be triggered. I thought the getter was smart for this tbh.